web/static/js/components/import_export_tray.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import CopyToClipboard from 'react-copy-to-clipboard'
+
+class ImportExportTray extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      pageContents: this.props.pageContents,
+      pageSettings: this.props.pageSettings,
+      stringifiedExportData: '',
+      stringifiedImportData: '',
+      exportDataButtonText: 'Copy to Clipboard'
+    }
+
+    this.onExportCopy = this.onExportCopy.bind(this)
+    this.stringifyExportData = this.stringifyExportData.bind(this)
+    this.onImportDataPaste = this.onImportDataPaste.bind(this)
+    this.onImportApply = this.onImportApply.bind(this)
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({
+      pageContents: nextProps.pageContents,
+      pageSettings: nextProps.pageSettings,
+      stringifiedExportData: '',
+      stringifiedImportData: '',
+      exportDataButtonText: 'Copy to Clipboard'
+    })
+  }
+
+  componentDidMount () {
+    this.stringifyExportData()
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (this.state.stringifiedExportData !== '') return
+
+    this.stringifyExportData()
+  }
```

Comment:
@jamonholmgren - the reason why I did it this way is because the CopyToClipboard wrapper requires that the textarea/input is already filled with data when you click on the button. If we were to do it your way, serialization would finish after the textarea has been copied. 

https://github.com/infinitered/thesis-phoenix/pull/94/files#diff-a4ba5f910195e5d928a4420e1ce3c3bdR116